src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -0,0 +1,340 @@
+     */
+    @Override
+    public boolean addAll(final int index,
+            final Collection<? extends T> collection) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Removes all of the elements from this list.
+     */
+    @Override
+    public void clear() {
+        for (int i = 0; i < this.index; i++) {
+            this.data[i] = null;
+        }
+        this.index = 0;
+    }
+
+    /**
+     * Checks if this list contains the specified element.
+     *
+     * @param element the element.
+     * @return true if this list contains the specified element.
+     */
+    @Override
+    public boolean contains(final Object element) {
+        for (int i = 0; i < this.index; i++) {
+            if (this.data[i].equals(element)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get the element at the specified position in this list..
+     *
+     * @param index the index.
+     * @return the element at the specified position in this list.
+     */
+    @Override
+    public T get(final int index) {
+        checkIndex(index);
+        return this.data[index];
+    }
+
+    /**
+     * Returns the index of the first occurrence of the specified element in
+     * this list, or -1 if this list does not contain the element.
+     *
+     * @param element the element
+     * @return the index
+     */
+    @Override
+    public int indexOf(final Object element) {
+        for (int i = 0; i < this.index; i++) {
+            if (this.data[i].equals(element)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * @param element the element.
+     * @return the index of the last occurrence of the specified element in this
+     *         list, or -1 if this list does not contain the element.
+     */
+    @Override
+    public int lastIndexOf(final Object element) {
+        for (int i = this.index - 1; i >= 0; i--) {
+            if (this.data[i].equals(element)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Checks if list empty.
+     *
+     * @return true if this list contains no elements.
+     */
+    @Override
+    public boolean isEmpty() {
+        return this.index == 0;
+    }
+
+    /**
+     * Replaces the element at the specified position in this list with the
+     * specified element.
+     *
+     * @param index   the index.
+     * @param element the element.
+     * @return the element previously at the specified position.
+     */
+    @Override
+    public T set(final int index, final T element) {
+        checkIndex(index);
+        final T prevElment = get(index);
+        this.data[index] = element;
+        return prevElment;
+    }
+
+    /**
+     * Returns the number of elements in this list.
+     */
+    @Override
+    public int size() {
+        if (this.index > Integer.MAX_VALUE) {
+            return Integer.MAX_VALUE;
```

Comment:
1- thought need to check with MX_VALUE to not exceed it.
2- don't know if there will be a size greater than MAX_VLAUE.
3- x = MAX_Integer + 1 = 2147483647 + 1.

Reply:
3- what is the value ??!! I know the maxValue is

Reply:
will be Integer.MIN_VALUE

Reply:
ok, it will be Integer.MIN_VALUE (-Ve value). Why does it becomes -Ve value ?
The explanation :- 

Let's imagine the integer is 3 bits 
NOTE :// ( Integer is 4 bytes = 32 bits. 3 BITS JUST FOR IMAGINATION TO MAKE EXPLANATION EASY.)
000 --> 0
001 --> 1
010 --> 2
011 --> 3

How will we represent -vs integers here ? [Hints : read 2's compliment ]
Read about 2s complement this : https://www.cs.cornell.edu/~tomf/notes/cps104/twoscomp.html#intro
After reading, please answer how will we represent -1, -2 , -3 , -4 if we IMAGINE integer is 3 bits. 

DO SAME example above if we imagine integer is 5 bits.
Then 
RETURN BACK TO REAL LIFE when the integer is 32 bits (4 bytes)  
how to represent 0, 1, 10, MAX_INTEGER, NEGATIVE INTEGER in memory  (32 bits ) ?
Also Calculate in binray MAX_INT + 1, what will be the value of it in binary ??